inbound_test.go

```diff
@@ -143,6 +143,41 @@ func TestInboundConnection_CallOptions(t *testing.T) {
 	})
 }
 
+func TestCallOptionsPropogated(t *testing.T) {
+	const handler = "handler"
+
+	giveCallOpts := CallOptions{
+		Format:          JSON,
+		CallerName:      "test-caller-name",
+		ShardKey:        "test-shard-key",
+		RoutingKey:      "test-routing-key",
+		RoutingDelegate: "test-routing-delegate",
+	}
+
+	var gotCallOptions *CallOptions
+
+	testutils.WithTestServer(t, nil, func(t testing.TB, ts *testutils.TestServer) {
+		ts.Register(HandlerFunc(func(ctx context.Context, inbound *InboundCall) {
+			gotCallOptions = inbound.CallOptions()
+			require.NotNil(t, gotCallOptions)
+
+			err := raw.WriteResponse(inbound.Response(), &raw.Res{})
+			assert.NoError(t, err, "write response failed")
+		}), handler)
+
+		ctx, cancel := NewContext(testutils.Timeout(time.Second))
+		defer cancel()
+
+		call, err := ts.Server().BeginCall(ctx, ts.HostPort(), ts.ServiceName(), handler, &giveCallOpts)
+		require.NoError(t, err, "could not call test server")
+
+		_, _, _, err = raw.WriteArgs(call, nil, nil)
+		require.NoError(t, err, "could not write args")
+
```

Comment:
we should probably read the response just to make sure the call is "completed" successfully. otherwise the test cleanup logic will probably wait for a timeout and will run slower

Reply:
`raw.WriteArgs` already reads both arg2 and arg3 is there something else I need to read?

https://github.com/uber/tchannel-go/blob/b1db3fe904fde57d507ecb4c4caf068dbce1a0f1/raw/call.go#L59

Reply:
Oh cool, I did not name this function well, thanks :)